### ISSUE TITLE
Create an explicit grab interaction for the book and zoomed-in book

### ISSUE THREAD

**Issue by terracoda** [phetsims/friction#126]
Knowing what to do with the book interaction. To paraphrase @emily-phet , the sim is just so simple, it becomes kind of flat. The book and the zoomed-in book don’t stand out enough. 

We think adding a button interaction will make the interaction more clear and easier to understand, and also make the interactions stand out more.

We want to try a version of the sim with button interactions.

Please implement new alerts from 
- [x] #124 and 
- [x] #125 first, and then create a branch for a version with two buttons.

The branched version will required some adjustments to the PDOM structure and content:
1. Structure Changes:
 - two new buttons in the Play Area: “Grab Chemistry book” and “Grab zoomed-in Chemistry book”
 - the div with application role will be hidden until one of the buttons is pressed

2. I'll make adjustments to the PDOM content to support the button interaction. 
 - [x] possible tweaks to intro paragraph and interaction cue
 - [x] tweaks to help text

These changes will all be very similar to BASE’s Grab Yellow Balloon button and interaction.

**Comment by terracoda** [comment]
@zepumph, I'll make the content and structure changes in the morning and assign to you when I think it is all clear.

**Comment by terracoda** [comment]
@zepumph and I discussed all the green highlights in the design document. @zepumph will implement the grab interactions as best he can with a close eye on how this interaction is in Balloons and Static Electricity.

While discussing these additions/changes, I realized I would need alerts for the grabbed state and released states. I will add this to the design document and make note where in this issue.

**Comment by terracoda** [comment]
> Is it ok that the grab helpText in in between the two buttons? If the first button it grabbed, 

@zepumph, great question. In BASE, we have the text twice, one for each, but in that sim the Green balloon is added dynamically (along with all its parts).
 
I'm just going to try it out and see.

**Comment by terracoda** [comment]
@zepumph, I am confused by your mark-up example.

What is the difference between
``` html
 < chemistry book help text>
<grab help text>
```
When you refer to <grab help text> what content are talking about?

I generally like to put explicit help text for an interaction after the interaction in case the user can just figure it out on their own what to do.

The hint in the opening scene, I think of as a hint rather than explicit help. Though really, it is all hints and it is all help.

**Comment by terracoda** [comment]
I see. What is needed now in the PDOM is the `<grab help text>`. It can go between the buttons or after both. It may be that it works better after both buttons.

Once a button is pressed (i.e., book or zoomed-in book is grabbed), the aria-role description alone "move in 4 directions", might not be enough to get them rubbing. 

I'll do a cognitive walk-through for the interaction in the design document and post a link once that is ready.

We will want to delivery one or two time some combination of the hints we have already created, "Rub fast or slow", "Move down to rub harder" on the first grab and then we will want o drastically shorten the grab alert on subsequent grabs after successful rubbing. 

This is what we did for BASE's grab button, though I we do not use aria-roledescription to deliver any of the information. This is our time to refine the custom grab implementation to see we can make it a better experience.

**Comment by zepumph** [comment]
I'm confused by the above comment, https://github.com/phetsims/friction/issues/126#issuecomment-431261129. I have implemented this grab help text, see the a11y view:
![image](https://user-images.githubusercontent.com/6856943/47259642-be512680-d458-11e8-869f-049839b659b1.png)

>We will want to delivery one or two time some combination of the hints we have already created, "Rub fast or slow", "Move down to rub harder" on the first grab and then we will want o drastically shorten the grab alert on subsequent grabs after successful rubbing.

Let me know what you decide and please assign me an issue to do that.

**Comment by terracoda** [comment]
Ooops, @zepumph, there are unchecked checkboxes in https://github.com/phetsims/friction/issues/126#issuecomment-430451004

Regarding the 4th checkbox question

>  Is it ok that the grab helpText in in between the two buttons? If the first button it grabbed, it looks like:
```<grab chemistry book>
  <chemistry book>
  < chemistry book help text>
<grab help text>
<grab zoomed in chemistry book>
```
After successful interaction the grab helptext is not needed, so I don't think it needs to be added to the PDOM. Does that make sense?

**Comment by terracoda** [comment]
I think the above comment also answers checkbox item 3. 
>  the book helptext is only shown when the macro book grabbed, not both. It is displayed only when you have grabbed the macro book.

I think the helptext, should actually show up in the alert, and not actually in the PDOM. Similar help, "Look for grab buttons. Once grabbed, ...." already exists in the PDOM.

@zepumph does the <chemistry book helptext> have to be inserted into the PDOM in order for it to be added to the alert on the initial grab?

**Comment by terracoda** [comment]
@zepumph, the only remaining question I have for this issue is what I asked in https://github.com/phetsims/friction/issues/126#issuecomment-433744379

> Does the <chemistry book helptext> have to be inserted into the PDOM in order for it to be added to the alert on the initial grab?

I'm thinking, not. For example, I don't think we add the initial move hint for the balloon in BASE to the PDOM. I think it might just be part of the aria-live alert on grab, but I am not 100% sure.

**Comment by terracoda** [comment]
Regarding your first checkbox question:
> shift tab will also go back to the button, kinda weird, but it is because we aren't hiding the buttons when focusing on the application div

It's ok that shift-tab goes back to the button, essentially releasing the grabbed book. This is how the balloon grab works in BASE, too. Tabbing away to the Zoomed-in book or Reset All button will also release whatever is grabbed.

A screen reader user will get the name of the newly focused item (the Grab button) and also likely some indication of a screen reader mode switch (e.g. I get a click in VoiceOver).

**Comment by terracoda** [comment]
Assigning back to you @zepumph to answer question in comment https://github.com/phetsims/friction/issues/126#issuecomment-433910617

**Comment by zepumph** [comment]
@terracoda I'm going to clarify just to be sure that I don't make a mistake:

The grab button help text should be in the PDOM until the first time a successful grab is made, on either book. Then we can remove that help text from the PDOM? I'm talking about:
`Look for grab buttons. Once grabbed, use letter keys W, A, S, or D to move book or zoomed-in book up, left, down, or right.`

The chemistry book help text should be deleted:
`Use arrow keys, or letter keys W, A, S, or D to move book or zoomed-in book up, left, down, or right.`

If that is incorrect, can we please talk next week? I feel like that will be simpler than the back and forth that could come.

**Comment by terracoda** [comment]
@zepumph, I forgot to assign back to you. Please re-assign to me if you have more questions.

**Comment by terracoda** [comment]
Also, since we are fully embracing the grab button interaction, I will clean up the design document asap, so it is less confusing.

**Comment by zepumph** [comment]
From meetings today:

* We need to get the verbose grab alert until we successfully "move"
* Treat "first time" alert separately for each book view/successful move. So successfully grabbing and moving the macro view won't change the fact that you will still hear the verbose alert on first grab of the zoomed-in book.

(for @terracoda)
* Add a verbose alert for when books are touching, but it's the first time grabbing THIS book view, i.e. a verbose, first-time, alert for "rub fast or slow".
* Remove the "Chemistry book rests lightly" because it is may be redundant since it is also in the PDOM.


Over to you @terracoda, please let me know what you decide, and I will implement everything together.

**Comment by terracoda** [comment]
@zepumph, as discussed needing 4 initial alerts in our meeting. I've updated the design document with these:

1. Initial grab when books are not touching (page load and after a reset):
   - Grabbed. Lightly on Physics book. Use W, A, S, or D keys to move book. Space to release.
2. Grab after successful interaction when books are not touching:
   - Grabbed. Lightly on Physics book. Move down to rub harder.
3. Initial grab (i.e. before successful interaction) when books are touching:
   - Grabbed. Rub fast or slow with A or D keys. Space to release.
4. Grab after successful interaction and when books are touching:
   - Grabbed. Rub fast or slow.

I decided to only mention A and D keys in alert # 3, as using W is not useful and using s won't do anything. 

I left all 4 keys in for alert # 1 for consistency with PDOM and if the user presses W they will get a hint to go down.

I think these might work well.

**Comment by terracoda** [comment]
@zepumph, I forgot to assign back to you.

**Comment by terracoda** [comment]
@zepumph, I do not hear the alerts above in https://github.com/phetsims/friction/issues/126#issuecomment-436439984 yet. I am going to remove the label "Ready to review" and change it to dev ally.

**Comment by zepumph** [comment]
@terracoda, I have completed what was asked of me above in master. Note that because of https://github.com/phetsims/scenery-phet/issues/421 there is some wonkyness with the grab button keyboard navigation, for example spacebar isn't supported yet (please use enter), and some edge cases won't cancel out of the "grabbed" mode correctly, so you may have to press escape to be safe more than you want. 

Otherwise I think you will be able to test this issue sufficiently. Please report issues with the above noted alerts here, and anything else you see about the grab button over in that issue. Thanks!

**Comment by terracoda** [comment]
@zepumph, what you have implemented sounds great!

I realize now that I did not define "successful interaction" and I might have over-thought alert number 3. I only hear it if I first use the mouse to put the books together, and then put keyboard focus on the book and grab it.

I am happy with the alerts as-is and I will open a new issue if we need any further adjustments. This issue is long enough!

**Comment by zepumph** [comment]
>I only hear it if I first use the mouse to put the books together, and then put keyboard focus on the book and grab it.

You can also hear it if you grab one of the book views, move the books to touch, and then highlight the other view and grab it for the first time. Is this acceptable?

**Comment by terracoda** [comment]
@zepumph, yes, that is more than acceptable. 

However, I just noticed that after a lot of successful interaction with the zoomed-in book, I hopped over to the book and I got alert 3.

New related question, **how does alert 3 get turned off?**

I think defining successful interaction will help.
Is it helpful if I define successful interaction as successful creation of friction in the model with either the book or zoomed-in book?

Then, after successful interaction, a user does not hear alerts 1 and 3 on either the book or zoomed-in book, and only alerts 2 and 4. Does that make good logical sense?

And just an FYI, I opened https://github.com/phetsims/a11y-research/issues/125 to determine if I need to be cuing the alternative keys so prominently.

**Comment by zepumph** [comment]
>I think defining successful interaction will help.

Your definition makes sense. Right now the definition of "success" is just "grabbing" a book view once. Since now each book view is a different alert system, this can happen for each independently,

I will look into updating this so that having a "successful" interaction effects both grab buttons. 

When a single successful interaction occurs, no matter what book view it was made with, neither book view will give the "before successful interaction" alert (alerts 1 or 3 above).

@terracoda could you please specify (with an estimate) about where on the thermometer the temperature needs to get to to constitute an interaction as "successfuly". Then I will fix this mistake of mine.

**Comment by terracoda** [comment]
@zepumph, a very small amount of friction is fine, just enough friction to cause the first jiggling alert for example. It's possible just a couple of horizontal moves left or right should work.

**Comment by zepumph** [comment]
I have updated the grab alerts for your "successful interaction" definition above.  @terracoda please review.

**Comment by terracoda** [comment]
@zepumph, I think you got it! I'm closing this mega issue :-)
